src/Controls/src/Core/Page/NavigatedFromEventArgs.cs

```diff
@@ -1,15 +1,30 @@
 ﻿#nullable disable
 using System;
+using Microsoft.Maui.Controls.Internals;
 
 namespace Microsoft.Maui.Controls
 {
+	internal enum NavigationType
+	{
+		Push,
+		Pop,
+		PopToRoot,
+		Insert,
+		Remove,
+		PageSwap, // Not a final name
+		PlaceHolder
```

Comment:
These 2 are "interesting"... They are fine for this PR or are you expecting an extra commit?

Reply:
I'll revisit before merging
In a perfect world these all get made public once we discuss this issue https://github.com/dotnet/maui/issues/21814

But, not sure if we'll have time for net9 unfortunately